src/Native/System.Native/pal_io.cpp

```diff
@@ -793,3 +794,17 @@ extern "C" int32_t Write(int32_t fd, const void* buffer, int32_t bufferSize)
     assert(count >= -1 && count <= bufferSize);
     return static_cast<int32_t>(count);
 }
+
+extern "C" int32_t GetWindowSize(WinSize* windowSize)
+{
+    assert(windowSize != nullptr);
+
+    int error = ioctl(1/*STDOUT_FILENO*/, TIOCGWINSZ, windowSize);
```

Comment:
> I am not sure how to do that

See https://github.com/dotnet/corefx/blob/master/src/Native/configure.cmake for a variety of examples.